### PR TITLE
Misc theme fixes

### PR DESCRIPTION
**Pull request by SychO9**
Some misc fixes for issues reported by **Antechinus**, I might push a few more commits so this is a draft for now, I'll try to keep it short though since there are many other pull requests which edit the CSS code.

---

**Changes:**
- Improvements for accessibility (add focus effects with hover, allow accessing certain list items, visual improvements)
- Code Improvements and simplifications.
- Fixes for some wonky elements.
- Removing unnecessary code.
- Visual improvements.

Much of it is probably not noticeable, but it's there ;)

### CODE REVIEW

**Comment by StealthWombat** [comment]
ETA: My mistake on this. I missed a bit. All good.

Sorry, but #5812 contains an error that will break functionality.

See: https://www.simplemachines.org/community/index.php?topic=569844.msg4032888#msg4032888

**Comment by StealthWombat** [comment]
Ok, further to the ramblings over at SMF, here's some proposed code that makes the styling consistent across all li's and a's.

It styles the checkbox li to match the anchors inside the other li's, and gives an indication of checkbox focus in Firefox and its derivatives. There is no actual hover styling on the checkbox li, as I found that didn't add anything to the visuals and wasn't worth the extra code. This code also fixes the funny looks with a .buttonrow .button on hover or focus next to an active button.

Has been tested in Firefox, Pale Moon, Chrome, Opera and IE11 on a W7 box and appears pixel perfect in all of them at standard zoom (checkbox li height will be slightly out at zoom levels of 200% or greater, but I doubt anyone with vision that bad will be worried).

Sources/Display.php (adds a class to the li)
```
		'quickmod' => array(
			'class' => 'inline_mod_check',
			'id' => 'in_topic_mod_check_'. $output['id'],
			'custom' => 'style="display: none;"',
			'content' => '',
			'show' => !empty($options['display_quick_mod']) && $options['display_quick_mod'] == 1 && $output['can_remove'],
		)
```

index.css
```
/* Styles for the standard button lists.
------------------------------------------------------- */
.buttonlist, .buttonrow, .pagelinks {
	z-index: 100;
	padding: 5px 0 5px 0;
	margin: 5px 0 5px 0;
}
a.button, .quickbuttons>li>a, .inline_mod_check {
	display: inline-block;
	padding: 0 8px;
	background: linear-gradient(to bottom, #fff 0%, #e2e9f3 70%);
	color: #444;
	font-size: 0.85em;
	line-height: 2em;
	text-transform: uppercase;
	text-decoration: none;
	height: calc(2em + 2em * (0.9 - 0.85)); /* "input" font size minus ".button" font size */
	border: 1px solid #ddd;
	border-right: 1px solid #bbb;
	border-bottom: 1px solid #aaa;
	border-radius: 3px;
	box-shadow: 1px 1px 1px #ddd;
}
.pagesection .button {
	color: #346;
}
.button:hover, .button:focus,
.quickbuttons>li>a:hover, .quickbuttons>li>a:focus {
	color: #222;
	cursor: pointer;
	border: 1px solid #ddd;
	border-left: 1px solid #bbb;
	border-top: 1px solid #aaa;
	box-shadow: -1px -1px 2px #ddd, -1px -2px 4px #ddd inset;
}
.button:hover, .button:focus {
	color: #af6700;
}
/* the active one */
.button.active {
	background: #557ea0;
	color: #fff;
	font-weight: bold;
	border: 1px solid #558080;
	text-shadow: 0 0 1px #000;
}
.button.active:hover, .button.active:focus {
	color: #ffc187;
	background: #557ea0;
	box-shadow: none;
}
/* In a .buttonrow, the buttons are joined together */
.buttonrow {
	margin: 0 5px;
}
.buttonrow .button {
	display: table-cell;
	border-radius: 0;
}

.buttonrow .button.active + .button:hover,
.buttonrow .button.active + .button:focus {
	box-shadow: -1px -2px 4px #ddd inset;
}
.buttonrow .button:first-child {
	border-radius: 3px 0 0 3px;
}
.buttonrow .button:last-child {
	border-radius:  0 3px 3px 0;
}
/* in a titlebg, the buttonlist is of small height */
.titlebg .buttonlist {
	margin: 0;
	padding: 0;
}
.under_message {
	display: flex;
	flex-wrap: wrap;
}
.likes_list {
	flex: 0 0 auto;
}
/* The quick buttons */
.quickbuttons {
	flex: 1 0 auto;
	text-align: right;
	font-size: 0.75rem;
}
#recent .quickbuttons {
	margin: 0;
}
.quickbuttons>li {
	position: relative;
	display: inline-block;
	margin: 0 -.2rem 0 0;
}
.quickbuttons>li>a, .inline_mod_check {
	display: block;
	height: auto;
	padding: 0 4px;
	color: #222;
	line-height: 1.4375rem;
	border: 1px solid #ddd;
	border-right: 1px solid #bbb;
	border-bottom: 1px solid #aaa;
	/* This is a case where shorthand hex
	/* is best for a box-shadow. It always
	/* overlays a blank background, so there
	/* is no benefit in the longer rgba syntax. */
	box-shadow: 1px 1px 1px #ddd;
	border-radius: 0;
}
.quickbuttons>li:first-child>a {
	border-radius: 4px 0 0 4px;
}
.quickbuttons>li:last-child>a, .inline_mod_check {
	border-radius: 0 4px 4px 0;
}
.inline_mod_check {
	height: 1.5625rem;
}
.moderationbuttons_check {
	position: relative;
	top: -1px;
}
.moderationbuttons_check:focus {
	box-shadow: 0 0 4px #499dd8;
}
```

**Comment by SychO9** [comment]
Awesome :D I'll take a look and push changes soon, thank you

**Comment by StealthWombat** [comment]
Just thought of something. This first bit was just copied from the current file, and I didn't pay attention to it:

```
/* Styles for the standard button lists.
------------------------------------------------------- */
.buttonlist, .buttonrow, .pagelinks {
	z-index: 100;
	padding: 5px 0 5px 0;
	margin: 5px 0 5px 0;
}
```

Obviously the margins and padding can be done in shorthand notation, but it may even be possible to ditch one. At a glance, padding: 10px 0; seems just as good as 5px padding and 5px margin (will test on local to make sure).

The other thing is that offhand I can't see any obvious reason why they would need a z-index declared. They're just basic buttons, and the z-index of the parent should be fine for them. Will test that too.

Edit: Have taken a look around the GUI, and I can't see any difference if it's done like this:

```
.buttonlist, .buttonrow, .pagelinks {
	padding: 10px 0;
}
```

There might be some edge case I'm missing, but AFAICT at the moment that code seems fine.

**Comment by StealthWombat** [comment]
Ok, the thread over at SMF is getting a bit convoluted. There's a range of suggested tweaks there, so I'll review and test all my current ideas for the .button and .quickbuttons classes, and get them into a more easily digested format. I'm sure the code can be less verbose overall and the looks can be a bit better. 

However, there is one simple tweak that I think is worthwhile: the active button on the main menu and other .dropmenu strips. It has a text-shadow which is pretty gruesome. There's a story there, and it's to do with web accessibility standards.

White text on an orange background does not meet even WCAG AA standards. It's nowhere close to passing, not even for large text (see: https://webaim.org/resources/contrastchecker/). The SMF team had it in their heads that the active button should be SMF orange and the text should be white. So I got sneaky and threw in a text-shadow in alpha, to give it some hope of being readable for a wide range of people while the project managers could still believe their button was SMF orange.

Unfortunately the current text-shadow is gruesome, and it's probably the original one I put there, so I feel obliged to suggest a fix. Like this:

```
/* Level 1 active button. */
.dropmenu a.active, #top_info a.active {
	color: #fff;
	font-weight: bold;
	border-color: #f49a3a;
	background: linear-gradient(#ffae14, #f5a100);
	text-shadow: 0 0 1px #000;
}
```
That, IMO, gives enough definition to make the button readable for most people, but looks a lot tidier than the current text-shadow. Would also suggest the same on hover and focus for all top level .dropmenu buttons.
```
.menu_nav>li>a:hover {
	text-shadow: 0 0 1px #000;
}
```

**Comment by SychO9** [comment]
Yup I agree with the shadow change, looks cleaner now.

I've also pushed other fixes, one of which adds the focus effect on the top_info menu items.

File: Sources/Profile-View.php [hunk 1]
```diff
@@ -695,6 +695,7 @@ function showAlerts($memID)
 				'icon' => 'move',
 			),
 			'quickmod' => array(
+    			'class' => 'inline_mod_check',
```
Comment (live627) [inline Sources/Profile-View.php:745]: needs extra tab here

**Comment by pouvik** [comment]
tested this myself, installs fine, gives forum a slightly crisper look, and causes no issues on install.

**Comment by MissAllSunday** [comment]
Hi @SychO9 can you please resolve the conflicts and give us an idea if you consider this to be done?

**Comment by SychO9** [comment]
I'll revisit this for RC4, I believe this should come after the other theme related pull requests are merged.

**Comment by SychO9** [comment]
btw I might end up reverting some of the design changes in this pull request, they are not bug fixes and because they are design changes there were a few spots where it didn't work out well/needs more tweaking to look better, which I don't believe is good to do at this stage, so I'll most likely clean this to keep fixes only.

**Comment by SychO9** [comment]
I believe this should be ready now, the only thing I undid was the very noticeable button design changes. Let me know if you'd like me to redo that, I just thought it best to keep the same design at this point and focus on the fixing/improvement aspect of this PR.

EDIT: Updated pr description.

**Comment by MissAllSunday** [comment]
Thanks! yeah the main idea is to focus on fixes rather than complete re-designing.

**Review by MissAllSunday — APPROVED** [review]
